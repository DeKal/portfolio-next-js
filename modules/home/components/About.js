import React from 'react'
import styled from 'styled-components/macro'
import Container from 'react-bootstrap/Container'
import CardHeader from '~/modules/home/components/about/CardHeader'
import CardInfo from '~/modules/home/components/about/CardInfo'

const About = () => (
  <Section>
    <Container>
      <Card>
        <CardHeader />
        <CardInfo />
      </Card>
    </Container>
  </Section>
)

const Section = styled.section`
  position: relative;
  margin-top: ${props => props.theme.aboutMarginTop};
`

const Card = styled.div`
  box-shadow: 2px 2px 4px -2px ${props => props.theme.colors.colorDark};
  z-index: 10;
  position: relative;
`
export default About
