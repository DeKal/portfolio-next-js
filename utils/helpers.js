import { INIT_SELECTED_PAGE } from '~/consts/initState'
import $ from 'jquery'

export const initHashLocation = () => {
  if (location.hash) {
    const offset = $(location.hash).offset()
    window.scrollTo(offset.left, offset.top)
  }
}

export const isSafari = () => {
  var userAgent = navigator.userAgent.toLowerCase()
  if (userAgent.indexOf('safari') != -1) {
    // Safari agent will not have Chrome
    if (userAgent.indexOf('chrome') > -1) {
      return false
    } else {
      return true
    }
  }
}

export const setJqueryScrollEvent = () => {
  $('a').on('click', function(event) {
    // Make sure this.hash has a value before overriding default behavior
    if (this.hash) {
      // Prevent default anchor click behavior
      event.preventDefault()
      // Store hash
      var hash = this.hash

      $('html, body').animate(
        {
          scrollTop: $(hash).offset().top
        },
        400,
        function() {
          // Add hash (#) to URL when done scrolling (default click behavior)
          window.location.hash = hash
        }
      )
    } // End if
  })
}

export const cleanUpScrollEvent = () => {
  $('a').unbind()
}

export const getSelectedSection = () => {
  if (location.hash.length !== 0) {
    const selectedPage = location.hash.substr(1)
    return selectedPage
  }
  return INIT_SELECTED_PAGE
}

export const isChristmasDay = () => {
  const today = new Date()
  const month = today.getMonth() + 1
  const isDecember = month == 12
  const day = today.getDate()
  const isChristMasTime = 20 <= day <= 30

  return isDecember && isChristMasTime
}

export const isNewYearComing = () => {
  const today = new Date()
  const day = today.getDate()
  const isFirstTwoWeek = 1 <= day <= 16
  const month = today.getMonth() + 1
  const isJanuary = month == 1

  return isJanuary && isFirstTwoWeek
}
