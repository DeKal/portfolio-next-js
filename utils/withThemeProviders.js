import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import defaultTheme from '~/views/theme'

export const shallowWithTheme = (children, theme = defaultTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

export const mountWithTheme = (children, theme = defaultTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}
