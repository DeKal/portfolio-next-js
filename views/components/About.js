import React from 'react'
import { ABOUT } from '~/consts/pages'
import styled from 'styled-components/macro'
import CardHeader from '~/views/components/about/CardHeader'
import CardInfo from '~/views/components/about/CardInfo'

const About = () => (
  <Container id={ABOUT} className="container">
    <Card className="about-card">
      <CardHeader />
      <CardInfo />
    </Card>
  </Container>
)

const Container = styled.section`
  margin-top: -200px;
`

const Card = styled.div`
  box-shadow: 2px 2px 4px -2px ${props => props.theme.colors.colorDark};
  float: none;
  margin: 0 auto;
`
export default About
