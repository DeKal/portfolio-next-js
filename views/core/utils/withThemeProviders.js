import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import homeTheme from '~/theme/theme'
import baseTheme from '~/theme/baseTheme'

/* istanbul ignore next */
export const shallowWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

/* istanbul ignore next */
export const shallowWithDivingTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children).dive()
}

/* istanbul ignore next */
export const mountWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}

/* istanbul ignore next */
export const shallowWithBaseTheme = (children, theme = baseTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}
