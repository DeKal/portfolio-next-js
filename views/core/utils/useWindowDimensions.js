import { useState, useEffect } from 'react'
import { getWindowDimensions } from '~/views/core/utils/helpers'

/* istanbul ignore next */
const useWindowDimensions = () => {
  const [windowDimensions, setWindowDimensions] = useState(
    getWindowDimensions()
  )

  useEffect(() => {
    /* istanbul ignore next */
    function handleResize() {
      setWindowDimensions(getWindowDimensions())
    }

    window.addEventListener('resize', handleResize)
    /* istanbul ignore next */
    return () => window.removeEventListener('resize', handleResize)
  }, [])

  return windowDimensions
}

export default useWindowDimensions
