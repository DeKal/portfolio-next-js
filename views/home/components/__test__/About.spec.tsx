import * as React from 'react'
import { mountWithTheme } from '~/utils/withThemeProviders'
import About from '~/views/home/components/About'

describe('About', () => {

  it('should render without throwing an error', function () {
    const wrap = mountWithTheme(<About />)
    expect(wrap).toMatchSnapshot()
  })

})