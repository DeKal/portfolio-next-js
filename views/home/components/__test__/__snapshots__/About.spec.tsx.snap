// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About should render without throwing an error 1`] = `
<About>
  <About__Section>
    <section
      className="About__Section-sc-1wp1oz6-0 OSnyC"
    >
      <Anchor
        id="About"
        top={-150}
      >
        <span
          className="Anchor-sc-6rgj1q-0 kAqcsG"
          id="About"
        />
      </Anchor>
      <Container
        fluid={false}
      >
        <div
          className="container"
        >
          <About__Card>
            <div
              className="About__Card-sc-1wp1oz6-1 gLfMwY"
            >
              <LocalizedComponent>
                <CardHeader
                  addTranslation={[Function]}
                  addTranslationForLanguage={[Function]}
                  ignoreTranslateChildren={false}
                  initialize={[Function]}
                  languages={Array []}
                  renderToStaticMarkup={false}
                  setActiveLanguage={[Function]}
                  translate={[Function]}
                >
                  <CardHeader__Header>
                    <header
                      className="CardHeader__Header-qscxnh-0 gYYlMW"
                    >
                      <CardHeader__MainText>
                        <h2
                          className="CardHeader__MainText-qscxnh-1 hJuEVu"
                        >
                          <Translate
                            id="About me"
                          >
                            <WrappedTranslate
                              context={
                                Object {
                                  "activeLanguage": undefined,
                                  "addTranslation": [Function],
                                  "addTranslationForLanguage": [Function],
                                  "defaultLanguage": undefined,
                                  "ignoreTranslateChildren": false,
                                  "initialize": [Function],
                                  "languages": Array [],
                                  "renderToStaticMarkup": false,
                                  "setActiveLanguage": [Function],
                                  "translate": [Function],
                                }
                              }
                              id="About me"
                            >
                              Missing translationId: About me for language: \${ languageCode }
                            </WrappedTranslate>
                          </Translate>
                        </h2>
                      </CardHeader__MainText>
                      <CardHeader__SubText>
                        <h4
                          className="CardHeader__SubText-qscxnh-2 gBRNDd"
                        >
                          <Translate
                            id="About Me: Subtext"
                          >
                            <WrappedTranslate
                              context={
                                Object {
                                  "activeLanguage": undefined,
                                  "addTranslation": [Function],
                                  "addTranslationForLanguage": [Function],
                                  "defaultLanguage": undefined,
                                  "ignoreTranslateChildren": false,
                                  "initialize": [Function],
                                  "languages": Array [],
                                  "renderToStaticMarkup": false,
                                  "setActiveLanguage": [Function],
                                  "translate": [Function],
                                }
                              }
                              id="About Me: Subtext"
                            >
                              Missing translationId: About Me: Subtext for language: \${ languageCode }
                            </WrappedTranslate>
                          </Translate>
                        </h4>
                      </CardHeader__SubText>
                    </header>
                  </CardHeader__Header>
                </CardHeader>
              </LocalizedComponent>
              <LocalizedComponent>
                <CardInfo
                  addTranslation={[Function]}
                  addTranslationForLanguage={[Function]}
                  ignoreTranslateChildren={false}
                  initialize={[Function]}
                  languages={Array []}
                  renderToStaticMarkup={false}
                  setActiveLanguage={[Function]}
                  translate={[Function]}
                >
                  <CardInfo__Container>
                    <div
                      className="CardInfo__Container-sc-1r6fj65-0 gwjRYO"
                    >
                      <p>
                        <Translate
                          id="About me: Card Info Verse 1"
                        >
                          <WrappedTranslate
                            context={
                              Object {
                                "activeLanguage": undefined,
                                "addTranslation": [Function],
                                "addTranslationForLanguage": [Function],
                                "defaultLanguage": undefined,
                                "ignoreTranslateChildren": false,
                                "initialize": [Function],
                                "languages": Array [],
                                "renderToStaticMarkup": false,
                                "setActiveLanguage": [Function],
                                "translate": [Function],
                              }
                            }
                            id="About me: Card Info Verse 1"
                          >
                            Missing translationId: About me: Card Info Verse 1 for language: \${ languageCode }
                          </WrappedTranslate>
                        </Translate>
                      </p>
                      <CardInfo__Quote>
                        <blockquote
                          className="CardInfo__Quote-sc-1r6fj65-1 fsoSGG"
                        >
                          <Translate
                            id="About me: Card Info Quote"
                          >
                            <WrappedTranslate
                              context={
                                Object {
                                  "activeLanguage": undefined,
                                  "addTranslation": [Function],
                                  "addTranslationForLanguage": [Function],
                                  "defaultLanguage": undefined,
                                  "ignoreTranslateChildren": false,
                                  "initialize": [Function],
                                  "languages": Array [],
                                  "renderToStaticMarkup": false,
                                  "setActiveLanguage": [Function],
                                  "translate": [Function],
                                }
                              }
                              id="About me: Card Info Quote"
                            >
                              Missing translationId: About me: Card Info Quote for language: \${ languageCode }
                            </WrappedTranslate>
                          </Translate>
                        </blockquote>
                      </CardInfo__Quote>
                      <p>
                        <Translate
                          id="About me: Card Info Verse 2"
                        >
                          <WrappedTranslate
                            context={
                              Object {
                                "activeLanguage": undefined,
                                "addTranslation": [Function],
                                "addTranslationForLanguage": [Function],
                                "defaultLanguage": undefined,
                                "ignoreTranslateChildren": false,
                                "initialize": [Function],
                                "languages": Array [],
                                "renderToStaticMarkup": false,
                                "setActiveLanguage": [Function],
                                "translate": [Function],
                              }
                            }
                            id="About me: Card Info Verse 2"
                          >
                            Missing translationId: About me: Card Info Verse 2 for language: \${ languageCode }
                          </WrappedTranslate>
                        </Translate>
                      </p>
                      <CardInfo__Status>
                        <p
                          className="CardInfo__Status-sc-1r6fj65-2 eKYGrj"
                        >
                          <CardInfo__Bold>
                            <span
                              className="CardInfo__Bold-sc-1r6fj65-3 dYwJPy"
                            >
                              Current Status:
                            </span>
                          </CardInfo__Bold>
                           
                          <Translate
                            id="About me: Current status"
                          >
                            <WrappedTranslate
                              context={
                                Object {
                                  "activeLanguage": undefined,
                                  "addTranslation": [Function],
                                  "addTranslationForLanguage": [Function],
                                  "defaultLanguage": undefined,
                                  "ignoreTranslateChildren": false,
                                  "initialize": [Function],
                                  "languages": Array [],
                                  "renderToStaticMarkup": false,
                                  "setActiveLanguage": [Function],
                                  "translate": [Function],
                                }
                              }
                              id="About me: Current status"
                            >
                              Missing translationId: About me: Current status for language: \${ languageCode }
                            </WrappedTranslate>
                          </Translate>
                        </p>
                      </CardInfo__Status>
                    </div>
                  </CardInfo__Container>
                </CardInfo>
              </LocalizedComponent>
            </div>
          </About__Card>
        </div>
      </Container>
    </section>
  </About__Section>
</About>
`;
